crates/shell/src/execute.rs

```diff
@@ -0,0 +1,44 @@
+use anyhow::Context;
+use deno_task_shell::{
+    execute_sequential_list, AsyncCommandBehavior, ExecuteResult, ShellPipeReader, ShellPipeWriter,
+    ShellState,
+};
+
+pub async fn execute_inner(text: &str, state: ShellState) -> anyhow::Result<ExecuteResult> {
+    let list = deno_task_shell::parser::parse(text);
+
+    let stderr = ShellPipeWriter::stderr();
+    let stdout = ShellPipeWriter::stdout();
+    let stdin = ShellPipeReader::stdin();
+
+    if let Err(e) = list {
```

Comment:
Is there any reason we are bailing on syntax errors? This causes shell to exit on syntax errors in interactive mode. 
Previously the error was being written to `stderr`.

Reply:
I think we might want to catch the error somewhere upstream!